src/elf/sym.rs

```diff
@@ -500,7 +500,10 @@ if_alloc! {
     impl<'a> Symtab<'a> {
         /// Parse a table of `count` ELF symbols from `offset`.
         pub fn parse(bytes: &'a [u8], offset: usize, count: usize, ctx: Ctx) -> Result<Symtab<'a>> {
-            let size = count * Sym::size_with(&ctx);
+            let size = count
+                .checked_mul(Sym::size_with(&ctx))
+                .ok_or(::error::Error::Malformed(format!("Too many ELF symbols (offset {:#x}, count {})",
```

Comment:
Are we 2018 in this crate ? We should update soon if not

Reply:
Yep, so rust 2018 + rustfmt?

Reply:
the only thing is we’ve had a really great story for supporting older compilers so I’d like to see what the Linux distros are shipping at the moment. If we can target a minimum rustc/ cargo combination which does 2018 that would be ideal. Need to investigate.